plugin-best-practices-plugin/src/main/kotlin/com/autonomousapps/internal/analysis/IssueListener.kt

```diff
@@ -316,3 +317,25 @@ internal class GetProjectListener : AbstractIssueListener() {
     return MethodNode.Metadata(isTaskAction)
   }
 }
+
+/** Invokes Eager APIs instead of Lazy ones.
+ * @see <a href="https://docs.gradle.org/current/userguide/lazy_configuration.html">Lazy Configuration</a>
+ * @see <a href="https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:old_vs_new_configuration_api_overview">Old vs New API Overview</a>
+ */
+internal class EagerApisListener : AbstractIssueListener() {
+
+  private val eagerApis = mapOf(
+    "org/gradle/api/tasks/TaskContainer" to setOf("all", "create", "getByName")
+  )
+
+  override fun computeIssues(): Set<Issue> = computeTraces().mapTo(HashSet()) { trace ->
+    EagerApiIssue("eagerApis", trace)
+  }
+
+  override fun isSuspectNode(graph: Graph<MethodNode>, methodNode: MethodNode): Boolean {
+    val methodOwner = methodNode.owner
+    val methodName = methodNode.name
+    return eagerApis.containsKey(methodOwner) && eagerApis.getValue(methodOwner)
+      .contains(methodName)
```

Comment:
could this be simplified to `eagerApis[methodOwner]?.contains(methodName)`?

Reply:
Oh that's much more readable, thanks